src/mame/konami/zr107.cpp

```diff
@@ -455,13 +455,16 @@ uint32_t zr107_state::ccu_r(offs_t offset, uint32_t mem_mask)
 			// Midnight Run polls the vertical counter in vblank
 			if (ACCESSING_BITS_24_31)
 			{
-				m_ccu_vcth ^= 0xff;
+				int32_t const vcth = m_ccu_vcth ^ 0xff;
+				if (!machine().side_effects_disabled())
+					m_ccu_vcth = vcth;
 				r |= m_ccu_vcth << 24;
 			}
 			if (ACCESSING_BITS_8_15)
 			{
-				m_ccu_vctl++;
-				m_ccu_vctl &= 0x1ff;
+				int32_t const vctl = (m_ccu_vctl + 1) & 0x1ff;
+				if (!machine().side_effects_disabled())
+					m_ccu_vctl = vctl;
 				r |= (m_ccu_vctl >> 2) << 8;
```

Comment:
Same here – previously the updated value was used.  Maybe you meant:
```c++
				int32_t const vctl = (m_ccu_vctl + 1) & 0x1ff;
				if (!machine().side_effects_disabled())
					m_ccu_vctl = vctl;
				r |= (vctl >> 2) << 8;
```

Reply:
I may suggest adding a todo stating ccu here is identical to k053252.cpp (here in zr107 it's driven by a cpld instead of the custom IC). Did some reversing of that area to figure that out.

No clue if it's possible to add the current k053252 device to zr107.

Reply:
@Hydreigon223 please don’t suggest adding speculation in comments.  Just leave this PR alone.